microk8s-resources/actions/enable.linkerd.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+# check if linkerd is already in the system.
+if [ ! -f "${SNAP_DATA}/bin/linkerd" ]; then
+
+  LINKERD_VERSION="${LINKERD_VERSION:-v2.2.1}"
+  echo "Fetching Linkerd2 version $LINKERD_VERSION."
+
+  sudo mkdir -p "$SNAP_DATA/bin"
+  LINKERD_VERSION=$(echo $LINKERD_VERSION | sed 's/v//g')
+  echo "$LINKERD_VERSION"
+  sudo "${SNAP}/usr/bin/curl" -L https://github.com/linkerd/linkerd2/releases/download/stable-${LINKERD_VERSION}/linkerd2-cli-stable-${LINKERD_VERSION}-linux -o "$SNAP_DATA/bin/linkerd"
+  sudo chmod uo+x "$SNAP_DATA/bin/linkerd"
+
+fi 
+
+echo "Enabling Linkerd2"
+
+# pod/servicegraph will start failing without dns
+"$SNAP/microk8s-enable.wrapper" dns
+
+"$SNAP_DATA/bin/linkerd" install "$*" | "$SNAP/kubectl" "--kubeconfig=$SNAP_DATA/credentials/client.config" apply -f -
```

Comment:
Could we be more precise in what arguments we allow? Seems we are passing all arguments passed but this may have bad implications.

Reply:
My initial thought was to allow only certain args to be passed, but then i realized that certain addons, in this case linkerd, allows several more args that can be passed while installing linkerd.  Such as mutual tls on the data plane... And maybe some other args in the future...